WcaOnRails/app/webpacker/lib/hooks/useHash.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+function getHashFromBrowserUrl() {
+  const urlHash = window.location.hash;
+  if (urlHash.length > 0) {
+    if (urlHash[0] === '#') {
+      return urlHash.slice(1);
+    }
+    return urlHash;
+  }
+  return null;
+}
+
+export default function useHash() {
+  const [hash, setHash] = React.useState(getHashFromBrowserUrl());
+
+  const onHashChange = React.useCallback(() => {
+    setHash(getHashFromBrowserUrl());
+  }, []);
```

Comment:
What does this callback do? The dependencies are empty.

Reply:
It was already there in react-use. I guess this is to avoid the function getting recreated while re-rendering.

Reply:
But memoizing doesn't make any sense when there is an empty dependency array.

@kr-matthews @coder13 We stole this hook from an open source library (didn't want to install several hundred KB and 7 transitive dependencies just for one hook...), can you figure out what this seemingly "empty" `useCallback` might be doing?

Reply:
It ensures that `onHashChange` points to the same function each re-render rather than pointing to a new (identical) function. This allows it to be used in _other_ dependency arrays without triggering them on every re-render.

Moving the definition into the event listener effect below (since it's only used there) would remove the need for this callback.

Reply:
I think keeping it outside is fine for the time being.